tools/astarte_import/lib/astarte/import/populatedb.ex

```diff
@@ -491,6 +491,83 @@ defmodule Astarte.Import.PopulateDB do
     end
   end
 
+  defp to_native_type(value_chars, :doublearray) do
+    with doublearray_string = to_string(value_chars),
+         {:ok, doublearray_list} <- Jason.decode(doublearray_string),
```

Comment:
We should not mix XML and JSON (see previous comment).
I'd suggest to take a look e.g. at how object `item`s are handled:
- When you start parsing the `item` element, you use the `:startElement` event to build the actual object item in memory, see [here](https://github.com/astarte-platform/astarte/blob/fe6af4f4d07bf3c663c7964823b76d4c0a99b0cd/tools/astarte_import/lib/astarte/import.ex#L260)
- When you're done with parsing the item, you can put it inside an accumulator, see [here](https://github.com/astarte-platform/astarte/blob/fe6af4f4d07bf3c663c7964823b76d4c0a99b0cd/tools/astarte_import/lib/astarte/import.ex#L291)
- When the full list (i.e. the actual value) is built, insert it in the database using a function (see [here](https://github.com/astarte-platform/astarte/blob/fe6af4f4d07bf3c663c7964823b76d4c0a99b0cd/tools/astarte_import/lib/astarte/import/populatedb.ex#L334). In this case, I think it would be actually possible not to write a new function, but just reuse [`got_end_of_value_fun`](https://github.com/astarte-platform/astarte/blob/fe6af4f4d07bf3c663c7964823b76d4c0a99b0cd/tools/astarte_import/lib/astarte/import/populatedb.ex#L304-L332).

Note that this approach would also allow for recycling the already existing `to_native_type` functions (as every element in the list would be parsed individually)

Reply:
Edit: `got_end_of_value_fun` would probably need some little adjustment to handle casting to the correct type

Reply:
Edit 2: the values are casted AFTER being parsed, just before being inserted in the database. So the `to_native_type` function for arrays would probably become just something like `Enum.map(array_value, fn element, type -> to_native_type(element, type) end)`